Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -286,6 +297,13 @@ santa_action_t SantaDecisionManager::GetFromCache(uint64_t identifier) {
         decision_cache->remove(identifier);
         return ACTION_UNSET;
       }
+    } else if (result == ACTION_RESPOND_ALLOW_TRANSITIVE) {
```

Comment:
It's because the the timestamps on the transitive rules only get updated when the rules are retrieved from the rules database.  If a transitive rule ends up in the decision cache and manages to stay in the cache for a long period of time, then its timestamp will never be updated even though the user might be running the binary it applies to fairly often.  Setting an expiration time on the transitive rules forces them to be periodically retrieved from the database, keeping its last access timestamp fresh.